src/Sentry/DynamicSamplingContext.cs

```diff
@@ -144,10 +161,18 @@ private DynamicSamplingContext(
             }
             items.Add("sample_rand", rand.ToString("N4", CultureInfo.InvariantCulture));
         }
+
+        if (replaySession?.ActiveReplayId is { } replayId)
+        {
+            // Overwrite any existing value - the DSC is simply used as a transport mechanism so that SDKs can
+            // communicate the replayId to Sentry Relay (SDKs don't need to propagate the replayId to each other).
```

Comment:
The SDKs do send the replay_id to each other. Through the `baggage` header. So that downstream SDKs can tag events with the replay_id (no 'joins done through trace-id).

The receiving SDK (the server) won't have a replay_id of itself, since it doesn't have a frontend. So we'll just use whatever id we receive in the incoming connection.

Reply:
Probably the comment needs rewording. 

We do propagate any upstream replay_id _as long as_ the current process hasn't started it's own replay session. On Android/iOS the chance of that are so remote as not to worry about. If we ever did session replay in a web based app, it might be more likely. 

In any case, if the current process has started it's own replay session then we overwrite any replay_id in the DSC, if there is one... since we don't have any other choice (you can only associate one replay_id with the DSC and you need to use the DSC to communicate the replay_id to Sentry). 

I'll see if I can find a better way to word the comment.